bliss/datasets/sdss.py

```diff
@@ -140,3 +140,164 @@ def get_from_disk(self, idx):
         pickle.dump(ret, field_dir.joinpath("cache.pkl").open("wb+"))
 
         return ret
+
+
+def _get_mgrid2(slen0, slen1):
+    offset0 = (slen0 - 1) / 2
+    offset1 = (slen1 - 1) / 2
+    x, y = np.mgrid[-offset0 : (offset0 + 1), -offset1 : (offset1 + 1)]
+    return torch.Tensor(np.dstack((y, x))) / torch.Tensor([[[offset1, offset0]]])
+
+
+def convert_mag_to_nmgy(mag):
+    return 10 ** ((22.5 - mag) / 2.5)
+
+
+def convert_nmgy_to_mag(nmgy):
+    return 22.5 - 2.5 * torch.log10(nmgy)
+
+
+def load_m2_data(
+    sdss_dir="../../sdss_stage_dir/",
+    hubble_dir="../hubble_data/",
+    slen=100,
+    x0=630,
+    x1=310,
+    f_min=1000.0,
+    border_padding=0,
+):
+    # returns the SDSS image of M2 in the r and i bands
+    # along with the corresponding Hubble catalog
+
+    #####################
+    # Load SDSS data
+    #####################
+    run = 2583
+    camcol = 2
+    field = 136
+
+    sdss_data = SloanDigitalSkySurvey(
+        sdss_dir,
+        run=run,
+        camcol=camcol,
+        field=field,
+        # returns the r and i band
+        bands=[2, 3],
+    )
+
+    # the full SDSS image, ~1500 x 2000 pixels
+    sdss_image_full = torch.Tensor(sdss_data[0]["image"])
+    sdss_background_full = torch.Tensor(sdss_data[0]["background"])
+
+    #####################
+    # load hubble catalog
+    #####################
+    hubble_cat_file = (
+        hubble_dir + "hlsp_acsggct_hst_acs-wfc_ngc7089_r.rdviq.cal.adj.zpt"
+    )
```

Comment:
Could put the stuff related to Hubble in it's own file, e.g. `datasets/hubble.py`?